accounts/abi/argument.go

```diff
@@ -160,6 +170,19 @@ func unpack(t *Type, dst interface{}, src interface{}) error {
 	return nil
 }
 
+// Unpack arguments into map
+func (arguments Arguments) unpackIntoMap(v map[string]interface{}, marshalledValues []interface{}) error {
+	// Make sure map is not nil
+	if v == nil {
+		return fmt.Errorf("abi: cannot unpack into a nil map")
+	}
+
+	for i, arg := range arguments.NonIndexed() {
```

Comment:
My understanding is that this needs to be done because the `log.Data` and `log.Topics` are [split up](https://github.com/ethereum/go-ethereum/blob/6b380a998d145923de5f77f2fbbcb4eafa7a4c88/accounts/abi/bind/base.go#L344) and need to be processed differently since the `data` is one chunk of bytes that needs to be decoded using [UnpackValues](https://github.com/ethereum/go-ethereum/blob/6b380a998d145923de5f77f2fbbcb4eafa7a4c88/accounts/abi/argument.go#L266) while the `topics` is a slice of hashes that need to be matched to their `abi.Argument` and processed by parseTopics/parseTopicsIntoMap. 

If this separation of processing was only done to impose the additional type safety on the topics, I wonder if we could concatenate `topics` bytes and `data` together and  unpack it all at once using UnpackValues? I will test this out.

Reply:
Doesn't work unfortunately, `UnpackValues` is built to operate only on non-indexed arguments